### ISSUE TITLE
TextEdit adds extra spacing in nested layouts

### ISSUE THREAD

**Issue by valadaptive** [emilk/egui#3709]
**Describe the bug**
I often nest left-to-right and right-to-left layouts inside each other in order to place elements along the ends.

When a `TextEdit` is the last element in such a layout, it adds extra padding that isn't present when it neighbors another widget directly:

![image](https://github.com/emilk/egui/assets/79560998/edab491c-b656-4910-bc44-afa0c7c416e6)

I've set the `item_spacing` to 0 to make the effect more clear.

In the top layout, the button on the right is placed first, using a right-to-left layout to make it the rightmost element. Then a left-to-right layout takes up the remaining space, and the label and textbox are placed inside it. This makes the textbox the last element within that nested layout, and there's thus a space between it and the button.

In the one below it, the label, textbox, and button are all part of the same left-to-right layout. This results in there being no extra spacing between them, as intended.

This nesting of layouts seems to be the only way to robustly handle resizing, so it's important for my use case:

![Peek 2023-12-16 02-44](https://github.com/emilk/egui/assets/79560998/2081ebb1-122a-4f35-914d-fb5926fda141)

**To Reproduce**

<details>
<summary>Demo snippet</summary>

```rust
ui.with_layout(egui::Layout::right_to_left(egui::Align::Center), |ui| {
    let _ = ui.button("Button");

    ui.with_layout(egui::Layout::left_to_right(egui::Align::Center), |ui| {
        ui.label("Label");
        let mut buf = String::from("This is some text");
        let _ = ui.text_edit_singleline(&mut buf);
    });
});

ui.with_layout(egui::Layout::left_to_right(egui::Align::Center), |ui| {
    ui.label("Label");
    let mut buf = String::from("This is some text");
    let _ = ui.text_edit_singleline(&mut buf);
    let _ = ui.button("Button");
});
```
</details>

**Expected behavior**
Both layouts should have the same spacing.

**Screenshots**
See above

**Additional context**
I've traced this problem back to [this commit](https://github.com/emilk/egui/pull/2649/commits/0c050b4cc92b05b4185e8e90c1d996567ac9898f#diff-e6f344a8826e9ea8eb014ac6547b935943468aa2e924e0be9263f50fd39f3926R407), which subtracts off a margin from the `TextEdit`'s size.

**Comment by toymil** [comment]
This happens when a TextEdit is last in a Grid cell too, you can actually see this directly in the egui online demo:

![Snipaste_2024-02-02_21-44-05](https://github.com/emilk/egui/assets/13289283/ae0fefc2-11be-493e-9807-d34b705d08aa)

**Comment by toymil** [comment]
This is fixed in #4005 (v0.26.1)
